deltachat-ios/Helper/ImageFormat.swift

```diff
@@ -0,0 +1,64 @@
+import Foundation
+import UIKit
+import SDWebImage
+
+enum ImageFormat: String {
+    case png, jpg, gif, tiff, webp, heic, bmp, unknown
+}
+
+extension ImageFormat {
+    static func get(from data: Data) -> ImageFormat {
+        switch data[0] {
+        case 0x89:
+            return .png
+        case 0xFF:
+            return .jpg
+        case 0x47:
+            return .gif
+        case 0x49, 0x4D:
+            return .tiff
+        case 0x52 where data.count >= 12:
+            let subdata = data[0...11]
+
+            if let dataString = String(data: subdata, encoding: .ascii),
+                dataString.hasPrefix("RIFF"),
+                dataString.hasSuffix("WEBP") {
+                return .webp
+            }
+
+        case 0x00 where data.count >= 12 :
+            let subdata = data[8...11]
+
+            if let dataString = String(data: subdata, encoding: .ascii),
+                Set(["heic", "heix", "hevc", "hevx"]).contains(dataString) {
+                return .heic
+            }
+
+        case 0x42 where data.count >= 2 :
+            if data[1] == 0x4D {
+                return .bmp
+            }
+
+        default:
+            break
+        }
+        return .unknown
```

Comment:
where does this list come from?
maybe add a link to the source (if any).

Reply:
https://en.wikipedia.org/wiki/List_of_file_signatures

Reply:
I'll can add that as a comment as well

Reply:
yip, i would add that link as a comment.

Reply:
done